week4/src/components/Result.tsx

```diff
@@ -36,22 +39,18 @@ export default function UserInfo() {
           ></DeleteBtn>
           <ProfilePic src={userGithubInfo?.avatar_url}></ProfilePic>
           <ProfileInfo>
-            <ProfileDetailInfo>
-              <ProfileDetailCargory>following </ProfileDetailCargory>
-              <ProfileDetailNum>{userGithubInfo?.following}</ProfileDetailNum>
-            </ProfileDetailInfo>
-
-            <ProfileDetailInfo>
-              <ProfileDetailCargory>follower </ProfileDetailCargory>
-              <ProfileDetailNum>{userGithubInfo?.followers}</ProfileDetailNum>
-            </ProfileDetailInfo>
-
-            <ProfileDetailInfo>
-              <ProfileDetailCargory>repo</ProfileDetailCargory>
-              <ProfileDetailNum>
-                {userGithubInfo?.public_repos}
-              </ProfileDetailNum>
-            </ProfileDetailInfo>
+            {["following", "follwer", "repo"].map((info) => (
+              <ProfileDetailInfo key={info}>
+                <ProfileDetailCargory>{info}</ProfileDetailCargory>
+                <ProfileDetailNum>
+                  {info === "following"
+                    ? userGithubInfo?.following
+                    : info === "follower"
+                    ? userGithubInfo?.followers
+                    : userGithubInfo?.public_repos}
+                </ProfileDetailNum>
+              </ProfileDetailInfo>
+            ))}
```

Comment:
옵셔널 체이닝이 많아서 한번 분리해봤어! 
이게 맘에 안들면 사실 tsx파일 하나더 만들어서 컴포넌트로 분리시켜도뎁니다! 
```suggestion js
const UserData = (userGithubInfo: iUserInfo) => {
    const { avatar_url, following, followers, public_repos } = userGithubInfo;
    return (
      <>
        <ProfilePic src={avatar_url}></ProfilePic>
        <ProfileInfo>
          {["following", "follwer", "repo"].map((info) => (
            <ProfileDetailInfo key={info}>
              <ProfileDetailCargory>{info}</ProfileDetailCargory>
              <ProfileDetailNum>
                {info === "following"
                  ? following
                  : info === "follower"
                  ? followers
                  : public_repos}
              </ProfileDetailNum>
            </ProfileDetailInfo>
          ))}
        </ProfileInfo>
      </>
    );
  };
  
  return (
    <>
      {loading ? (
        <Loading />
      ) : (
        <ResultFrame>
          <DeleteBtn
            type="button"
            onClick={() => navigate(-1)}
            value="❎"
          ></DeleteBtn>
          {userGithubInfo && UserData(userGithubInfo)}
        </ResultFrame>
      )}
    </>
  );
```